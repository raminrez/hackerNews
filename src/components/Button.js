import React from "react";

export default function Button({ onClick, className = "", children }) {
  return (
    <button onClick={onClick} className={className} type="button">
      {children}
    </button>
  );
}
