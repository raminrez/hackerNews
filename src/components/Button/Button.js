import React from "react";
import PropTypes from "prop-types";

export default function Button({ onClick, className, children }) {
  return (
    <button onClick={onClick} className={className} type="button">
      {children}
    </button>
  );
}

Button.defaultProps = {
  className: ""
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};
