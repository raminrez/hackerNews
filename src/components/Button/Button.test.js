import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import Button from "./Button";

describe("Button", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button>Do Somethings</Button>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test("should have valid snapshot", () => {
    const component = renderer.create(<Button>Do Somethings</Button>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
