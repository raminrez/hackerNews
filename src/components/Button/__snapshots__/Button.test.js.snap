// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should have valid snapshot 1`] = `
<button
  className=""
  type="button"
>
  Do Somethings
</button>
`;
